packages/gestalt/src/Flex.js

```diff
@@ -0,0 +1,123 @@
+// @flow strict
+import React, { Children, type Node } from 'react';
+import PropTypes from 'prop-types';
+import FlexItem from './FlexItem.js';
+import styles from './Flex.css';
+import wrapWithComponent from './utils/wrapWithComponent.js';
+import { buildStyles } from './boxTransforms.js';
+import {
+  AlignContentPropType,
+  AlignItemsPropType,
+  AlignSelfPropType,
+  DimensionPropType,
+  DirectionPropType,
+  FlexPropType,
+  GapPropType,
+  JustifyContentPropType,
+  OverflowPropType,
+  type AlignContent,
+  type AlignItems,
+  type AlignSelf,
+  type Dimension,
+  type Direction,
+  type Flex as FlexType,
+  type Gap,
+  type JustifyContent,
+  type Overflow,
+} from './boxTypes.js';
+
+type Props = {|
+  alignContent?: AlignContent,
+  alignItems?: AlignItems,
+  alignSelf?: AlignSelf,
+  children?: Node,
+  direction?: Direction,
+  flex?: FlexType,
+  gap?: Gap,
+  height?: Dimension,
+  justifyContent?: JustifyContent,
+  maxHeight?: Dimension,
+  maxWidth?: Dimension,
+  minHeight?: Dimension,
+  minWidth?: Dimension,
+  overflow?: Overflow,
+  width?: Dimension,
+  wrap?: boolean,
+|};
+
+const allowedProps = [
+  'alignContent',
+  'alignItems',
+  'alignSelf',
+  'children',
+  'direction',
+  'flex',
+  'height',
+  'justifyContent',
+  'maxHeight',
+  'maxWidth',
+  'minHeight',
+  'minWidth',
+  'overflow',
+  'width',
+  'wrap',
+];
+
+export default function Flex({
+  alignItems = 'center',
+  children: childrenProp,
+  direction = 'row',
+  gap = 0,
+  justifyContent = 'center',
+  ...rest
+}: Props): Node {
+  const children = gap
+    ? Children.map(childrenProp, (child, index) => {
+        if (child === null || child === undefined) {
+          return null;
+        }
+        return wrapWithComponent({
```

Comment:
@dangerismycat would there be a way to make this an actual React component?

Reply:
Sure. Does that gain us anything? The avalanche of Flow errors I had to deal with while adapting this code means we're definitely not losing type safety. 😅 It accepts a generic type that it seems to be able to infer even without setting explicitly.

Reply:
@dangerismycat mostly making it easier to read through and see that it returns a React Element. This is a NIT, not a must have (since we can always change it later)

Reply:
Cool cool, we can refactor later since it's internal.